mmf/datasets/processors/frcnn_processor.py

```diff
@@ -0,0 +1,173 @@
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.import copy
+ """
+import os
+import sys
+from io import BytesIO
+from typing import List
+
+import numpy as np
+import omegaconf
+import requests
+import torch
+import torch.nn.functional as F
+from mmf.common.registry import registry
+from mmf.datasets.processors.processors import BaseProcessor
+from PIL import Image
+
+
+class ResizeShortestEdge:
+    def __init__(self, short_edge_length: List[int], max_size: int = sys.maxsize):
+        """
+        Args:
+            short_edge_length (list[min, max])
+            max_size (int): maximum allowed longest edge length.
+        """
+        self.interp_method = "bilinear"
+        self.max_size = max_size
+        self.short_edge_length = short_edge_length
+
+    def __call__(self, imgs: List[torch.Tensor]):
+        img_augs = []
+        for img in imgs:
+            h, w = img.shape[:2]
+            # later: provide list and randomly choose index for resize
+            size = np.random.randint(
+                self.short_edge_length[0], self.short_edge_length[1] + 1
+            )
+            if size == 0:
+                return img
+            scale = size * 1.0 / min(h, w)
+            if h < w:
+                newh, neww = size, scale * w
+            else:
+                newh, neww = scale * h, size
+            if max(newh, neww) > self.max_size:
+                scale = self.max_size * 1.0 / max(newh, neww)
+                newh = newh * scale
+                neww = neww * scale
+            neww = int(neww + 0.5)
+            newh = int(newh + 0.5)
+
+            if img.dtype == np.uint8:
+                pil_image = Image.fromarray(img)
+                pil_image = pil_image.resize((neww, newh), Image.BILINEAR)
+                img = np.asarray(pil_image)
+            else:
+                img = img.permute(2, 0, 1).unsqueeze(0)  # 3, 0, 1)  # hw(c) -> nchw
+                img = F.interpolate(
+                    img, (newh, neww), mode=self.interp_method, align_corners=False
+                ).squeeze(0)
+            img_augs.append(img)
+
+        return img_augs
+
+
+@registry.register_processor("frcnn_preprocess")
+class FRCNNPreprocess(BaseProcessor):
+    def __init__(self, cfg: omegaconf.dictconfig.DictConfig, *args, **kwargs):
+        self.aug = ResizeShortestEdge(
+            [cfg.input.min_size_test, cfg.input.min_size_test], cfg.input.max_size_test
+        )
+        self.input_format = cfg.input.format
```

Comment:
You can use `.get` to get config values and then add a default value for user's sanity.